src/main/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildWrapper.java

```diff
@@ -878,17 +900,43 @@ public SauceConnectHandler call() throws AbstractSauceTunnelManager.SauceConnect
 
                 } else if (StringUtils.isBlank(key)) {
                     listener.getLogger().println("Access key not set, not starting Sauce Connect");
-
-                } else {
-                    Process process = sauceTunnelManager.openConnection(username, key, port, sauceConnectJar, options, listener.getLogger(), verboseLogging, sauceConnectPath);
                 }
-                return this;
             } catch (ComponentLookupException e) {
                 throw new AbstractSauceTunnelManager.SauceConnectException(e);
             }
+
+            if (maxRetries > 0) {
+                int retryCount = 0;
+                while (retryCount < maxRetries) {
+                    try {
+                        Process process = sauceTunnelManager.openConnection(username, key, port, sauceConnectJar, options, listener.getLogger(), verboseLogging, sauceConnectPath);
+                        return this;
+                    } catch (AbstractSauceTunnelManager.SauceConnectDidNotStartException e) {
+                        retryCount++;
+                        if (retryCount >= maxRetries) {
+                            throw new AbstractSauceTunnelManager.SauceConnectException(e);
+                        } else {
+                            listener.getLogger().println(String.format("Error launching Sauce Connect, trying %s time(s) more.", (maxRetries - retryCount)));
+                        }
+                        wait(retryWaitTime);
+                    }
+                }
+            } else {
+                Process process = sauceTunnelManager.openConnection(username, key, port, sauceConnectJar, options, listener.getLogger(), verboseLogging, sauceConnectPath);
+            }
+            return this;
+        }
+
+        private void wait(int seconds) {
+            try {
+                Thread.sleep(1000 * seconds);
+            } catch (InterruptedException e) {
```

Comment:
You are right, I did not consider that option. I have not worked with Jenkins plugins so much.
I tried it and it keeps on trying since the Interrupt is caught.

I changed the code to throw the InterruptedException when the wait is done, didn't like the idea of having a try-catch inside another one, but now it works as expected when the build is cancelled.